### DISCUSSION TITLE
Why are Floats not automatically converted? (but Integers are)

### DISCUSSION THREAD

**Discussion by sebnozzi** [ruby/ruby.wasm#211] [Q&A]
When passing parameters in a JavaScript call Integers and Strings are converted just fine; but Floats aren't.

Why is this so?

Consider this code:

```ruby
require "js"

JS.global[:window].call(:alert, 123) # works
JS.global[:window].call(:alert, 123.5) # FAILS
```

Instead, one has to resort to this:

```ruby
JS.global[:window].call(:alert, JS.eval("return 123.5")) # works
```

Is this intended? Am I missing something? (seems like an overkill to me)

Also see my other thread about not being able to call `to_js` to a Float value.

[Link in "try Ruby Playground"](https://try.ruby-lang.org/playground/#code=require+%22js%22%0A%0AJS.global%5B%3Awindow%5D.call(%3Aalert%2C+123)+%23+works%0AJS.global%5B%3Awindow%5D.call(%3Aalert%2C+123.5)+%23+FAILS%0A%0A%23+Workaround+(%3F)%0AJS.global%5B%3Awindow%5D.call(%3Aalert%2C+JS.eval(%22return+123.5%22))+%23+works%0A&engine=cruby-3.2.0)

**Comment by kateinoigakukun** [comment]
Closed issue https://github.com/ruby/ruby.wasm/issues/213
